### DISCUSSION TITLE
Automatically remove closed issues after a certain time

### DISCUSSION THREAD

**Discussion by azenMatt** [community/community#4256] [Projects and Issues]
I've set up my project board to track a backlog of work, and have a separate view for a current work cycle / sprint. Over time I end up with a lot of closed issues that I then have to manually remove.

It would be great if these items were auto-archived or filtered out after a certain amount of time.

**Comment by r002** [comment]
Hi @azenMatt -- coincidentally, this issue that you posted from two weeks ago is another item in my own project that I just knocked out [today](https://github.com/studydash/cards/issues/171).  While it'd be neat if GitHub natively supported `time-triggered` actions that are configurable from the GUI, in the interim, for anyone who's in this same situation, one workaround is spinning up a free GCP instance (or cloud provider of your choice) and scheduling a simple cron job to run at whatever time interval you need to invoke the GitHub apis to do your bidding (clean/archive/re-label issues, etc).

In my scenario, I keep a layer of `GitHub analytics` in Firestore for my project.  (GitHub issue stats like `streaks`-- the number of days a member consecutively contributes, for example).  But every night (12:01a ET), I needed a simple job that would hit up the GitHub API, reconcile/aggregate stats, and write this second-order data to Firestore.

So in a nutshell:

1. Write a simple event-driven function that does whatever GitHub tasks you need:

![image](https://user-images.githubusercontent.com/45280066/124521764-ae233600-ddbe-11eb-922a-478857cf292d.png)

2. Deploy it as cloud function on your free tier:

![image](https://user-images.githubusercontent.com/45280066/124521387-51734b80-ddbd-11eb-9c40-5d96811342be.png)

3. Schedule a cron job for it:

![image](https://user-images.githubusercontent.com/45280066/124521429-710a7400-ddbd-11eb-89ff-178e42f286c0.png)

This is how I keep our study group's `GitHub Dashboard` updated every night! 😀🎉🥳

![image](https://user-images.githubusercontent.com/45280066/124521707-8d5ae080-ddbe-11eb-9897-84ba01b34aec.png)

**Comment by nickhagen** [comment]
We use this github action to handle this: 
https://github.com/actions/stale

Using this, you can specify a stale period, and then it will automatically close stale issues and insert a last comment indicating that they should review/re-open their issue if they feel it should still be worked on.

I prefer this method as it gives me a lot of control with custom responses and status!

**Reply by nickhagen** [reply]
oof sorry about that, re-read your issue and i didn't realize the beta didn't drop off close issues!  I just realized because I had setup a new project to test this, it setup the closed status as 'Done' and they do not drop off the board ... that does feel bad!  Will upvote this request.

**Reply by nickhagen** [reply]
another possible implementation that i have seen is to allow you to just hide the closed column on Kanban or collapse it to a small column

**Comment by szeitlin** [comment]
I would love to at least be able to hide closed issues from project boards. I don't want to remove them, necessarily, as it's often useful to have the history there (at least until the project is finished).

**Reply by azenMatt** [reply]
you could add a saved filter to help with this, `-is:closed` would work
